### PR TITLE
feat: update to b2sdk 1.21.0

### PR DESCRIPTION
**Pull request by Marmelatze**
the new b2sdk has an abstract method list_bucket_names_ids

### CODE REVIEW

**Comment by ehossack** [comment]
Thanks for the pull! Sorry for the delay, busy week. Will take a look ASAP :)

File: django_backblaze_b2/cache_account_info.py [hunk 1]
```diff
@@ -186,6 +186,12 @@ def remove_bucket_name(self, bucket_name):
         self.cache.set("bucket_names", [n for n in self.cache.get("bucket_names", []) if n != bucket_name])
         self.cache.delete(_bucket_cachekey(bucket_name))
 
+    def list_bucket_names_ids(self) -> List[Tuple[str, str]]:
+        res = []
+        for name in self.cache.get("bucket_names"):
+            res.append((name, self.cache.get(_bucket_cachekey(name))))
+        return res
```
Comment (ehossack) [inline django_backblaze_b2/cache_account_info.py:193]: unlikely, but we should handle the null case here.

Also a nit, but maybe a slightly clearer name than `res`, e.g.

```
    def list_bucket_names_ids(self) -> List[Tuple[str, str]]:
        tuples = []
        for bucket_name in self.cache.get("bucket_names", []):
            bucket_id = self.cache.get(_bucket_cachekey(bucket_name))
            tuples.append((bucket_name, bucket_id))
        return tuples
```

I can make the change if you don't get to it.

Comment (Marmelatze) [reply django_backblaze_b2/cache_account_info.py]: Thanks for the feedback, I updated the PR

File: tests/test_cache_account_info.py [hunk 2]
```diff
@@ -172,10 +172,12 @@ def test_can_refresh_entire_bucket_name_cache():
     bucket_id_or_none = cacheAccountInfo.get_bucket_id_or_none_from_bucket_name("some-name")
     bucket_id_or_none2 = cacheAccountInfo.get_bucket_id_or_none_from_bucket_name("other-name")
     random_id_or_none = cacheAccountInfo.get_bucket_id_or_none_from_bucket_name("something-random")
+    name_ids = cacheAccountInfo.list_bucket_names_ids()
```
Comment (ehossack) [inline tests/test_cache_account_info.py:175]: sorry one more comment here, I think probably this warrants its own test similar to `test_get_bucket_id_when_bucket_name_set` and `test_get_bucket_id_when_bucket_name_not_set`

Exceeppttt, reading this test, I can't for the life of me figure out why the list is `[("some-name", "new-id")]` and not `[("some-name", "new-id"), ("something-random", "some-random-id")]` 

Can you explain why? 🤔

Comment (ehossack) [reply tests/test_cache_account_info.py:175]: ahhhhhhh there's a bug in this method, the cache for all names isn't updated 😅

Comment (ehossack) [reply tests/test_cache_account_info.py:175]: Fixed in #110 -- thanks for uncovering this, unintentionally 😛

Comment (ehossack) [reply tests/test_cache_account_info.py]: I re-added the modification. Seemed easier than going back and forth.

**Comment by ehossack** [comment]
Thanks for the contribution! I made the tests a bit more specific, rather than trying to go back and forth on the PR. Hope you don't mind :)

**Comment by ehossack** [comment]
errr... tests pass locally. Not sure how to get the CI pipeline to trigger. A todo for later :P

**Comment by Marmelatze** [comment]
No Problem, thanks for the fast merge. Would you mind making a new Release?

**Comment by ehossack** [comment]
Of course. I have some family stuff to take care of and will merge the other minor updates then release.

**Comment by ehossack** [comment]
3.2.0 has your change, 3.2.1 has the other (non-prod) dependency updates :)

**Pull request merged**